tests/unit/s2n_mem_usage_test.c

```diff
@@ -54,7 +54,7 @@
 #ifdef __FreeBSD__
     #define MEM_PER_CONNECTION 57
 #elif defined(__OpenBSD__)
-    #define MEM_PER_CONNECTION 71
+    #define MEM_PER_CONNECTION 75
```

Comment:
Do we have the reason for these memory differences documented anywhere? Could be nice to link those in a comment, because right now it fells all feels rather arcane

Reply:
And perhaps we should consider substantially increasing the `MAX_CONNECTION` from 1000 to >> 1000 to reduce the variability in this test?

Reply:
Both good questions --> I don't know the answers. @maddeleine ?

Reply:
On the second one I'm guessing that we don't want to blow up the time/memory usage of this test by too many orders of magnitude (x10 might be ok?).  If anything I'd like to dive deeper and understand where the variability is coming from first. @goatgoose mentioned that it might be a result of the VM since he wasn't able to replicate it in some other contexts. 

The goal of this test (as I understand it) is mainly to ensure that our memory usage stays capped. The OpenBSD platform is a relatively new edition to the CI and so raising this number feels like just part of the process of setting the ceiling in the first place.

Reply:
The test has failed, so either the test is wrong or the property under test is wrong. I agree that it's _probably_ something a little funky with the test, but a 50% increase in memory utilization feels large, so it seems prudent to make sure.

Reply:
I have a theory that I have yet to validate. Its possible that the increase in memory usage is related to Debug/Release build?

Next step add logging to always print the size computed and see if it is drastically smaller in Release than Debug.

Reply:
We should do an investigation of this high reported memory usage, but this should also not be blocking developers.

Reply:
Here is a tracking issue: #3795